provisioning/ubuntu-provision.sh

```diff
@@ -156,7 +156,8 @@ function install_asdf() {
   rm -f "${archive}"
 
   ## append to the system wide path variable, need to be seconded for docker in packer sources.pkr.hcl
-  sed -e "/^PATH/s/\"$/:\${asdf_install_dir}\/shims:\${asdf_install_dir}\/bin\"/g" -i /etc/environment
+  ## https://backreference.org/2010/02/20/using-different-delimiters-in-sed/index.html
+  sed --regexp-extended "s|^PATH=\(.*\)\"$|PATH=\1:${asdf_install_dir}/shims:${asdf_install_dir}/bin\"|" /etc/environment
```

Comment:
```suggestion
  sed --in-place --regexp-extended "s|^PATH=(.*)\"|PATH=\1:${asdf_install_dir}/shims:${asdf_install_dir}/bin\"|g" /etc/environment
```

Reply:
- Ref.: https://www.gnu.org/software/sed/manual/html_node/ERE-syntax.html , the `--regexp-extended` (same as `-r` and `-E`) means we must not escape special characters such as parenthesis (`(` / `)`) unless we want them literate. Our CI errors was due to no escape group in the search pattern: as such `\1` was unknown
- The `g` at the end of the sed argument has no influence: I've added it by reflex